subprojects/packagefiles/sqlitecpp/meson.build

```diff
@@ -0,0 +1,298 @@
+project(
+    'SQLiteCpp', 'cpp',
+    # SQLiteCpp requires C++11 support
+    default_options: ['cpp_std=c++11'],
+    license: 'MIT',
+    version: '3.2.1',
+)
+
+cxx = meson.get_compiler('cpp')
+
+## at best we might try to test if this code compiles
+## testing for compilers or platforms is not reliable enough
+## example: native clang on windows or mingw in windows
+unix_like_code = '''
+    #if defined(unix) || defined(__unix__) || defined(__unix)
+    // do nothing
+    #else
+    # error "Non Unix-like OS"
+    #endif
+'''
+unix_like = cxx.compiles(unix_like_code, name : 'unix like environment')
+
+mingw_64_env_code = '''
+    #if defined(__MINGW64__)
+    // do nothing
+    #else
+    # error "Non MinGW-W64 environment"
+    #endif
+'''
+mingw_64_env = cxx.compiles(mingw_64_env_code, name : 'MinGW-W64 environment')
+
+thread_dep = dependency('threads')
+# sqlite3 support
+sqlite3_dep = dependency(
+                'sqlite3',
+                fallback: ['sqlite3', 'sqlite3_dep']
+)
+
+sqlitecpp_incl = [ 
+    include_directories('include')
+]
+sqlitecpp_srcs = [
+    'src/Backup.cpp',
+    'src/Column.cpp',
+    'src/Database.cpp',
+    'src/Exception.cpp',
+    'src/Savepoint.cpp',
+    'src/Statement.cpp',
+    'src/Transaction.cpp',
+]
+sqlitecpp_args = [
+    '-Wall',
+]
+sqlitecpp_link = []
+sqlitecpp_deps = [
+    sqlite3_dep,
+    thread_dep,
+]
+## used to override the default sqlitecpp options like cpp standard
+sqlitecpp_opts = []
+
+## tests
+
+sqlitecpp_test_srcs = [
+    'tests/Column_test.cpp',
+    'tests/Database_test.cpp',
+    'tests/Savepoint_test.cpp',
+    'tests/Statement_test.cpp',
+    'tests/Backup_test.cpp',
+    'tests/Transaction_test.cpp',
+    'tests/VariadicBind_test.cpp',
+    'tests/Exception_test.cpp',
+    'tests/ExecuteMany_test.cpp',
+]
+sqlitecpp_test_args = []
+
+## samples
+
+sqlitecpp_sample1_srcs = [
+    'examples/example1/main.cpp',
+]
+sqlitecpp_sample2_srcs = [
+    'examples/example2/src/main.cpp',
+]
+
+# if not using MSVC we need to add this compiler arguments
+# for a list of MSVC supported arguments please check:
+# https://docs.microsoft.com/en-us/cpp/build/reference/compiler-options-listed-alphabetically?view=msvc-170
+if not (host_machine.system() == 'windows' and cxx.get_id() == 'msvc')
+    sqlitecpp_args += [
```

Comment:
could simplify with cc.get_supported_arguments.

Reply:
The flip side is that hardcoding it based on get_id() is faster because you don't need to run the compiler each time to check which ones are supported.

Reply:
I agree, it will be a faster option if the parameters are known in advance, as there are few arguments you can simply use this option, if in the future there were more arguments, perhaps it would be a better option to check if MSVC is used, looking on the bright side we can remove additional verification that is used only for MSVC.

Honestly I don´t know much about the meson implementation and if those values are cached for an specific compiler/setup as it could reduce setup times if they are checked across projects, but I think that is more dependent on implementation than project configuration, maybe a linter should be helpful for this cases(and for new users), as for the moment I think I will set `cxx.get_supported_arguments` as the meson.build in the project is not a 1-1 copy to the CMakeLists.txt anyways

Reply:
done

Reply:
It would only be cached within a project so that checking the same option again already knows if it's supported.